src/core/ScreenLockListenerDBus.cpp

```diff
@@ -51,6 +52,15 @@ ScreenLockListenerDBus::ScreenLockListenerDBus(QWidget *parent):
                 this, //receiver
                 SLOT(logindPrepareForSleep(bool)));
 
+    QString sessionId = QProcessEnvironment::systemEnvironment().value("XDG_SESSION_ID");
```

Comment:
I don't think this will open us to exploits, but I'd feel better if you validated this to be an integer.